### ISSUE TITLE
hi(●'◡'●)

### ISSUE THREAD

**Issue by jun0wanan** [qinzzz/Multimodal-Alignment-Framework#2]
Will you give us your train  feature ?  I want to train your code..

**Comment by jun0wanan** [comment]
test_detection_dict.json
train_imgid2idx.pkl

  this work you didn't give us . Your link is error.(You can try to  click ) >< Hope to give your reply


best wishes

**Comment by qinzzz** [comment]
Hi jun, sorry for the late reply. I have updated the link to test_detection_dict.json and uploaded my train features. May you try again?

**Comment by jun0wanan** [comment]
> Hi jun, sorry for the late reply. I have updated the link to test_detection_dict.json and uploaded my train features. May you try again?

您需要访问权限 ><

**Comment by jun0wanan** [comment]
> Hi jun, sorry for the late reply. I have updated the link to test_detection_dict.json and uploaded my train features. May you try again?

Your coding is very beautiful and great ! hope to run your code and hope to receive your reply !
Best wishes

**Comment by jun0wanan** [comment]
![Uploading image.png…]()

**Comment by qinzzz** [comment]
I am soooo sorry! I have fixed that.

**Comment by jun0wanan** [comment]
> I am soooo sorry! I have fixed that.

Hello, thank you for your prompt reply. Do you plan to open source Val Bbox? o(*￣▽￣*)ブ

val_detection_dict.json

**Comment by qinzzz** [comment]
Sure! https://drive.google.com/file/d/1KmyG0mghwydkb7pEwxDjItwZvNi_DRA4/view?usp=sharing. By the way, I am uploading val features and it may take a while.

**Comment by jun0wanan** [comment]
> Sure! https://drive.google.com/file/d/1KmyG0mghwydkb7pEwxDjItwZvNi_DRA4/view?usp=sharing. By the way, I am uploading val features and it may take a while.

wow, Thank you!! So happy ~ 
Do you plan to open source Val feat? ❤

val_features.hdf5

and then I can train modal ~ 
thank you your code 
It is really perfect ~

**Comment by qinzzz** [comment]
Sure, I have just uploaded it: https://drive.google.com/file/d/1W_m8QdSmEZfsLj-clGE7lPOJV6jJgNa_/view?usp=sharing

**Comment by jun0wanan** [comment]
> Sure, I have just uploaded it: https://drive.google.com/file/d/1W_m8QdSmEZfsLj-clGE7lPOJV6jJgNa_/view?usp=sharing

Thank you. I'm sorry to bother you 😊

**Comment by jun0wanan** [comment]
您好(●'◡'●)，我将您的代码作为入门学习pytorch的代码之一
我想请问一下，里面这个函数的操作
`def _encode_pk`
对于每个phrase 为什么 不进行mask掉pad呢？在里面计算 softmax的时候，因为我发现 还是有可能影响的，不是吗
`p_emb = torch.einsum('byq,byqd -> byd', q_max_norm_att, q_emb) `
希望得到您的回答

**Comment by qinzzz** [comment]
Yeah...I believe you are right. In the previous version 0 paddings were converted to -inf so I didn't mask the pad. Currently the softmax is smoothed by the 0s. It might be better to mask the padding and I will try to fix that. BTW, it's also my first pub so the code might not be perfect. Thank you for looking into my code!

**Comment by jun0wanan** [comment]
> Yeah...I believe you are right. In the previous version 0 paddings were converted to -inf so I didn't mask the pad. Currently the softmax is smoothed by the 0s. It might be better to mask the padding and I will try to fix that. BTW, it's also my first pub so the code might not be perfect. Thank you for looking into my code!

thank you your reple！ Your code is very clear, so it is also excellent I think~

**Comment by jun0wanan** [comment]
> Yeah...I believe you are right. In the previous version 0 paddings were converted to -inf so I didn't mask the pad. Currently the softmax is smoothed by the 0s. It might be better to mask the padding and I will try to fix that. BTW, it's also my first pub so the code might not be perfect. Thank you for looking into my code!

thank you your reply！ Your code is very clear, so it is also excellent I think~
